### ISSUE TITLE
Passing non-extendable event handlers to addEventListner causes exceptions

### ISSUE THREAD

**Issue by chrisinajar** [LogRocket/logrocket#17]
If you pass a method that is frozen (via `Object.preventExtensions` or similar) as the handler to `addEventListener` on any HTML node after logrocket has been initialized it results in an immediate exception and the event is not registered.

This causes errors with certain typescript implemented react libraries such as https://github.com/xiel/embla-carousel-wheel-gestures which use strict definitions for the shapes of their event listeners.

`TypeError: Cannot add property __lr_wrapper__, object is not extensible`

You can see this error is caused by the `wrap` [function](https://github.com/getsentry/sentry-javascript/blob/2be35b46184bffbafd384ccbd6b706eda83bbfe8/src/raven.js#L202).
```jsx
      // copy over properties of the old function
      for (var property in func) {
        if (hasKey(func, property)) {
          wrapped[property] = func[property];
        }
      }
      wrapped.prototype = func.prototype;

      func.__lr_wrapper__ = wrapped;
      // Signal that this function has been wrapped already
      // for both debugging and to prevent it to being wrapped twice
      wrapped.__lr__ = true;
      wrapped.__inner__ = func;

      return wrapped;
    }
```

This can be solved be either testing if the function is extensible first, or simply not having this function be in `'strict mode'`

**Comment by xiel** [comment]
Hey Logrocket,

I created a minimal reproduction case: https://codesandbox.io/s/logrocket-frozen-eventhandler-o7p7h?file=/src/index.js

I think I would be a good idea to not mutate objects you do not own ;) You could easily associate data with a function using a WeakMap or similar.

– Felix

**Comment by chrisinajar** [comment]
> Hey Logrocket,
> 
> I created a minimal reproduction case: https://codesandbox.io/s/logrocket-frozen-eventhandler-o7p7h?file=/src/index.js
> 
> I think I would be a good idea to not mutate objects you do not own ;) You could easily associate data with a function using a WeakMap or similar.
> 
> – Felix

I had the same thought too, but the shims for WeakMap in incompatible browsers rely on overriding `.indexOf`, so that wouldn't solve the exception in every case. They _also_ need to remove `'use strict'` to prevent the exception and make it a noop in incompatible scenarios.

**Comment by lexispike** [comment]
Issue should be fixed in [version 1.0.14](https://www.npmjs.com/package/logrocket/v/1.0.14) of the npm package!

**Comment by xiel** [comment]
Thank you @lexispike , can confirm 💯 
updated case: https://codesandbox.io/s/logrocket-frozen-eventhandler-forked-164z7?fontsize=14&hidenavigation=1&theme=dark
